broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java

```diff
@@ -34,6 +35,21 @@ public RaftPartitionFactory(final BrokerCfg brokerCfg) {
   }
 
   public RaftPartition createRaftPartition(final PartitionMetadata partitionMetadata) {
+    final var partitionDirectory =
```

Comment:
:thought_balloon: I see this is added for restore. Restore only needs the partitionId and the data directory, I think. It would be better if we don't have to create full raft partition for it.